tests/integration_tests/test_reproducibility.py

```diff
@@ -0,0 +1,182 @@
+        random_seed(int): random seed integer to use for test
+        second_seed_offset(int): zero to use same random seed for second test, non-zero to use a different
+            seed for the second run.
+
+    Returns: None
+    """
+    # define Ludwig model
+    model1 = LudwigModel(config=CONFIG, logging_level=logging.WARN)
+    training_statistics1, preprocessed_data1, _ = model1.train(
+        dataset=raw_dataset_fp, random_seed=random_seed, skip_save_progress=True, skip_save_processed_input=True
+    )
+
+    # invoke torch random functions
+    torch.manual_seed(random_seed + 1)
+    torch.rand((5,))
+
+    model2 = LudwigModel(config=CONFIG, logging_level=logging.WARN)
+    training_statistics2, preprocessed_data2, _ = model2.train(
+        dataset=raw_dataset_fp,
+        random_seed=random_seed + second_seed_offset,
+        skip_save_progress=True,
+        skip_save_processed_input=True,
+    )
+
+    # confirm data splits are reproducible
+    for i in range(3):
+        for k in preprocessed_data1[i].dataset:
+            if second_seed_offset == 0:
+                # same seeds should result in same output
+                assert np.all(preprocessed_data1[i].dataset[k] == preprocessed_data2[i].dataset[k])
+            else:
+                # non-zero second_seed_offset uses different seeds and should result in different output
+                with pytest.raises(AssertionError):
+                    assert np.all(preprocessed_data1[i].dataset[k] == preprocessed_data2[i].dataset[k])
+
+    # confirm reproducibility/non-reproducibility of results
+    if second_seed_offset == 0:
+        # same seeds should result in same output
+        assert training_statistics1 == training_statistics2
+    else:
+        # non-zero second_seed_offset uses different seeds and should result in different output
+        with pytest.raises(AssertionError):
+            assert training_statistics1 == training_statistics2
+
+
+@pytest.mark.parametrize("second_seed_offset", [0, 1])
+@pytest.mark.parametrize("random_seed", [1919, 31])
+def test_experiment(raw_dataset_fp: str, random_seed: int, second_seed_offset: int) -> None:
+    """Test reproducibility of experiment API.
+
+    Args:
+        raw_dataset_fp (str): file path for data to be used as part of this test
+        random_seed(int): random seed integer to use for test
+        second_seed_offset(int): zero to use same random seed for second test, non-zero to use a different
+            seed for the second run.
+
+    Returns: None
+    """
+    # define Ludwig model
+    model1 = LudwigModel(config=CONFIG, logging_level=logging.WARN)
+
+    evaluation_statistics1, training_statistics1, preprocessed_data1, _ = model1.experiment(
+        dataset=raw_dataset_fp, random_seed=random_seed, skip_save_processed_input=True
+    )
+
+    # invoke torch random functions
+    torch.manual_seed(random_seed + 1)
+    torch.rand((5,))
```

Comment:
Same question: Is it necessary to set these seeds manually in the test for the test to pass?